programs/dynamic-bonding-curve/src/instructions/creator/ix_transfer_pool_creator.rs

```diff
@@ -24,16 +27,54 @@ pub struct TransferPoolCreatorCtx<'info> {
     pub new_creator: UncheckedAccount<'info>,
 }
 
-pub fn handle_transfer_pool_creator(ctx: Context<TransferPoolCreatorCtx>) -> Result<()> {
+pub fn handle_transfer_pool_creator<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, TransferPoolCreatorCtx>,
+) -> Result<()> {
     let mut pool = ctx.accounts.virtual_pool.load_mut()?;
 
     let migration_progress = pool.get_migration_progress()?;
-    // avoid pool creator to do update between 2 periods
-    require!(
-        migration_progress == MigrationProgress::PreBondingCurve
-            || migration_progress == MigrationProgress::CreatedPool,
-        PoolError::NotPermitToDoThisAction
-    );
+    let config = ctx.accounts.config.load()?;
+    match migration_progress {
+        MigrationProgress::PreBondingCurve => {
+            // always work
+        }
+        MigrationProgress::CreatedPool => {
+            let migration_option = MigrationOption::try_from(config.migration_option)
+                .map_err(|_| PoolError::InvalidMigrationOption)?;
+            if migration_option == MigrationOption::MeteoraDamm {
+                // check if creator and partner has claim lp token and locked escrow
+                let migration_metadata_account = ctx
+                    .remaining_accounts
+                    .get(0)
+                    .ok_or_else(|| PoolError::InvalidAccount)?;
+                let migration_metadata_loader: AccountLoader<'_, MeteoraDammMigrationMetadata> =
+                    AccountLoader::try_from(migration_metadata_account)?; // TODO fix damm config in remaning accounts
```

Comment:
Why don't put it in the account context?

Reply:
Fixed comment
Because that function can work with any migration options, so proper fix should be stored migration progress in pool state, so later we add a new migration option, we can have a common place to check (no need metadata)
So I just temp check in remaning account